pkg/common/cns-lib/vsphere/virtualmachine.go

```diff
@@ -337,3 +337,38 @@ func (vm *VirtualMachine) IsInZoneRegion(ctx context.Context, zoneCategoryName s
 func GetUUIDFromProviderID(providerID string) string {
 	return strings.TrimPrefix(providerID, providerPrefix)
 }
+
+// GetSharedDatastoresForVMs returns shared datastores accessible to specified
+// nodeVMs list.
+func GetSharedDatastoresForVMs(ctx context.Context, nodeVMs []*VirtualMachine) ([]*DatastoreInfo, error) {
```

Comment:
Have we unit tested this function with empty nodeVMs[] ? 
If not, are the callers of this function checking for length of `sharedDatastores` returned by this function to avoid NPE

Reply:
yes the callers check for length of `sharedDatastores` and appropriately error out.